GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs

```diff
@@ -65,20 +69,126 @@ public CreateTestCaseRequestExtended BuildCaseRequest(Scenario scenario, IFeatur
                 WorkflowId = AddWorkflow(scenario, featureFile),
                 Description = AddDescription(scenario, featureFile),
                 Scenario = AddScenario(scenario, featureFile),
-                Tags = AddTags(scenario, featureFile)
+                Tags = AddTags(scenario, featureFile),
+                CustomFields = AddCustomFields(scenario, featureFile)
             },
             StepsAttachments = AddStepAttachments(scenario, featureFile)
         };
 
         return createTestCaseRequestExtended;
     }
 
+    private List<CustomFieldItem> AddCustomFields(Scenario scenario, IFeatureFile featureFile)
+    {
+        var result = new List<CustomFieldItem>();
+        AddFeatureCustomFiled(featureFile, result);
+        AddEpicCustomFiled(featureFile, result);
+        AddComponentCustomFiled(featureFile, scenario, result);
+        return result;
+    }
+
+    private void AddComponentCustomFiled(IFeatureFile featureFile, Scenario scenario, List<CustomFieldItem> result)
+    {
+        var settingsComponent = _allureTestOpsSettings.Component;
+
+        var allTags = GherkinHelper.GetAllTags(scenario, featureFile);
+        var componentTag = allTags.LastOrDefault(tag => tag.Name.Contains(TagsConstants.Component, StringComparison.InvariantCultureIgnoreCase));
+
+        if (componentTag is not null)
+        {
+            var componentField = GetField("Component");
+
+            var componentValues = componentTag.Name.Replace(TagsConstants.Component, "").Split(",");
+            foreach (var componentValue in componentValues)
+            {
+                AddCustomFiled(result, componentField, componentValue);    
+            }
+            
+            return;
+        }
+
+        if (settingsComponent is not null)
+        {
+            var componentField = GetField("Component");
+            AddCustomFiled(result, componentField, settingsComponent);
+        }
+    }
+
+    private CustomFieldSchemaContent GetField(string fieldName)
+    {
+        var componentField = CustomFieldSchema.FirstOrDefault(content => content.CustomField.Name.Equals(fieldName));
+
+        if (componentField is null)
+        {
+            Log.Error($"{fieldName} field is not found.");
+            _context.IsRunSuccessful = false;
+            return componentField;
+        }
+
+        var values = _allureClientWrapper.GetCustomFieldValues(componentField.CustomField.Id);
+
+        if (!_customFilesValues.ContainsKey(componentField.CustomField.Id))
+        {
+            _customFilesValues.Add(componentField.CustomField.Id, values.ToList());
+        }
+
+        return componentField;
+    }
+
+    private void AddCustomFiled(List<CustomFieldItem> result, CustomFieldSchemaContent customField, string cfValue)
+    {
+        var value = _customFilesValues[customField.CustomField.Id].FirstOrDefault(item => item.Name.Equals(cfValue));
```

Comment:
maybe, the _customFilesValues should be renamed into _customFieldValues?

Reply:
Done